infra/AuthStack.py

```diff
@@ -171,6 +171,89 @@ def _get_client_secret(
         return describe_cognito_user_pool_client.get_response_field(
             "UserPoolClient.ClientSecret"
         )
+    
+    def _create_policy(self) -> iam.PolicyDocument:
+        """
+        Create a policy for the KMS key to restrict access to the service.
+        """
+        return iam.PolicyDocument(
+            statements=[
+                iam.PolicyStatement(
+                    actions=[
+                        "kms:Encrypt",
+                        "kms:Decrypt",
+                        "kms:ReEncrypt*",
+                        "kms:CreateGrant",
+                        "kms:DescribeKey"
+                    ],
+                    effect=iam.Effect.ALLOW,
+                    resources=["*"],
+                    principals=[iam.ArnPrincipal("*")],
```

Comment:
What does it mean to allow any principal for this policy?

Reply:
AWS states it as "Allow access through AWS Secrets Manager for all principals in the account that are authorized to use AWS Secrets Manager"